NeoMathEngine/src/CPU/CpuMathEngineDnnConv.cpp

```diff
@@ -370,44 +356,35 @@ void CCpuMathEngine::blobConvolutionForwardAlgo1( const CCpuConvolutionDesc& des
 	float* outputTransposedData = GetRaw( stackBuffer.GetHandle() );
 	float* tempBlobData = outputTransposedData + outputTransposedDataSize;
 
-	NEOML_OMP_NUM_THREADS( curThreadCount )
-	{
-		int batchStart;
-		int batchCount;
-		int resultStart;
-		int resultCount;
-		if( OmpGetTaskIndexAndCount2D( source.ObjectCount(), result.Width(), batchStart, batchCount, resultStart, resultCount ) ) {
-			for( int batch = batchStart; batch < batchStart + batchCount; ++batch ) {
-				const int tempObjectIndex = source.ObjectCount() <= tempObjectCount ? batch : OmpGetThreadNum();
-				float* outputTransposedPtr = outputTransposedData + tempObjectIndex * outputTransposedDataObjectSize
-					+ resultStart * outputTransposedDataRowSize;
-				float* tempBlobPtr = tempBlobData + tempObjectIndex * tempBlobDataObjectSize
-					+ resultStart * tempBlobDataRowSize;
-
-				// Fill the temporary matrix
-				if( desc.DilationHeight > 1 || desc.DilationWidth > 1 ) {
-					createDilationTemporaryBlob( desc, sourceData, batch, resultStart, resultCount, tempBlobPtr );
-				} else {
-					createTemporaryBlob( desc, sourceData, batch, resultStart, resultCount, tempBlobPtr );
-				}
+	const int resultCount = result.Width();
+	const int batchCount = source.ObjectCount();
 
-				// Apply the filter to the temporary matrix
-				if( freeTermData != nullptr ) {
-					setVectorToMatrixRows( outputTransposedPtr, result.Height() * resultCount, outputChannels, freeTermDataRaw );
+	for( int batch = 0; batch < batchCount; ++batch ) {
+		float* const outputTransposedPtr = outputTransposedData + batch * outputTransposedDataObjectSize;
+		float* const tempBlobPtr = tempBlobData + batch * tempBlobDataObjectSize;
```

Comment:
Here the problem is similar to `outputTransposedPtr`

Reply:
fixed